inc/customizer/layouts.php

```diff
@@ -0,0 +1,499 @@
+				'sanitize_callback' => 'sanitize_html_class',
+				'transport'         => 'postMessage',
+			)
+		);
+
+		$wp_customize->add_control(
+			'layout',
+			array(
+				'label'       => esc_html__( 'Default Layout', 'primer' ),
+				'description' => esc_html__( 'All posts and pages on your site will use this layout by default.', 'primer' ),
+				'section'     => 'layout',
+				'settings'    => 'layout',
+				'type'        => 'radio',
+				'choices'     => $this->layouts,
+			)
+		);
+
+		$wp_customize->add_setting(
+			'full_width',
+			array(
+				'default'           => 0,
+				'sanitize_callback' => 'absint',
+			)
+		);
+
+		$wp_customize->add_control(
+			'full-width',
+			array(
+				'label'       => esc_html__( 'Header & Footer Width', 'primer' ),
+				'description' => esc_html__( 'Set the header and footer width to be fixed or fluid when viewed on large screens.', 'primer' ),
+				'section'     => 'layout',
+				'settings'    => 'full_width',
+				'type'        => 'radio',
+				'choices'     => array(
+					0 => esc_html__( 'Fixed', 'primer' ),
+					1 => esc_html__( 'Fluid', 'primer' ),
+				),
+			)
+		);
+
+	}
+
+	/**
+	 * Add layout class to body element on the front-end.
+	 *
+	 * @filter body_class
+	 * @since  1.0.0
+	 *
+	 * @param  array $classes
+	 *
+	 * @return array
+	 */
+	public function body_class( array $classes ) {
+
+		$classes[] = sanitize_html_class( sprintf( 'layout-%s', $this->get_current_layout() ) );
+		$classes[] = (bool) get_theme_mod( 'full_width' ) ? 'no-max-width' : null;
+
+		return array_filter( $classes );
+
+	}
+
+	/**
+	 * Check if a layout exists.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param  string $layout
+	 *
+	 * @return bool
+	 */
+	protected function layout_exists( $layout ) {
+
+		return isset( $this->layouts[ $layout ] );
+
+	}
+
+	/**
+	 * Return a post ID.
+	 *
+	 * @param  WP_Post|int $post (optional)
+	 *
+	 * @return string
+	 */
+	protected function get_post_id( $post = null ) {
+
+		return is_a( 'WP_Post', $post ) ? $post->ID : ( is_numeric( $post ) ? absint( $post ) : get_queried_object_id() );
+
+	}
+
+	/**
+	 * Return the layout override for a post.
+	 *
+	 * @param  WP_Post|int $post (optional)
+	 *
+	 * @return string
+	 */
+	protected function get_post_layout( $post = null ) {
+
+		return get_post_meta( $this->get_post_id( $post ), 'primer_layout', true );
+
+	}
+
+	/**
+	 * Return the global layout.
+	 *
+	 * @return string
+	 */
+	public function get_global_layout() {
+
+		return get_theme_mod( 'layout', $this->default );
+
+	}
+
+	/**
+	 * Return the current layout.
+	 *
+	 * @param  WP_Post|int $post (optional)
+	 *
+	 * @return string
+	 */
+	public function get_current_layout( $post = null ) {
+
+		$override = $this->get_post_layout( $post );
+		$layout   = ( $override ) ? $override : $this->get_global_layout();
+
+		return $this->layout_exists( $layout ) ? $layout : $this->default;
+
+	}
+
+}
+
+$GLOBALS['primer_customizer_layouts'] = new Primer_Customizer_Layouts;
```

Comment:
@jonathanbardo Just to give a little background on why I chose to assign the class to a global var. Layouts are a special case where helper functions in the global scope need to be available for theme developers when creating Child themes. The object itself if pretty lightweight and I believe this approach is better than instantiating a new object every time you want to call a helper method.